### PR TITLE
Update membership role filter to use new annotation

### PR DESCRIPTION
**Pull request by benjaminapetersen**
@jwforres @spadgett

### CODE REVIEW

**Comment by benjaminapetersen** [comment]
@enj fyi

File: app/scripts/services/membership/membership.js [hunk 1]
```diff
@@ -114,18 +114,9 @@ angular
...
-        // image-puller & image-pusher ok, other system: prob no
-        return (_.isEqual(item.metadata.name, 'system:image-puller')   ||
-                _.isEqual(item.metadata.name, 'system:image-pusher')   ||
-                _.isEqual(item.metadata.name, 'system:image-builder')  ||
-                _.isEqual(item.metadata.name, 'system:deployer')     ) ||
-                ! _.startsWith(item.metadata.name, 'cluster-') &&
-                ! _.startsWith(item.metadata.name, 'system:') &&
-                ! _.startsWith(item.metadata.name, 'registry-') &&
-                ! _.startsWith(item.metadata.name, 'self-');
+        return !annotation(item, 'systemOnly');
```
Comment (jwforres) [inline app/scripts/services/membership/membership.js]: you need to actually check the content of the annotation and compare it to the string 'true' in theory someone could set the annotation and set it to false

**Comment by jwforres** [comment]
@benjaminapetersen have you run the latest origin images and made sure this works as expected in combination with that?

**Comment by benjaminapetersen** [comment]
@jwforres yup, built it this morn

File: app/scripts/services/membership/membership.js (hunk 1 above)
Comment (benjaminapetersen) [inline app/scripts/services/membership/membership.js]: Right, & verified with @enj that is consistent with some other annotations.  So only `'true'` will be honored. Updated.

**Comment by jwforres** [comment]
@benjaminapetersen the travis failure is because your spec tests need to be updated

**Comment by benjaminapetersen** [comment]
fixed
[TEST]

File: test/spec/services/membership/membershipSpec.js [hunk 2]
```diff
@@ -87,27 +87,14 @@ describe('MembershipService', function() {
...
-        {metadata: {name: 'system:deployer'}},
-        // roles the filter will explicitly reject
-        {metadata: {name: 'cluster-magical-scary-role'}},
-        {metadata: {name: 'system:special-scary-role'}},
-        {metadata: {name: 'registry-ninja'}},
-        {metadata: {name: 'self-destruction'}},
-        // ignored
-        {metadata: {name: 'admin'}},
-        {metadata: {name: 'edit'}}
+        {metadata: {name: 'system-only-role', annotations: {'authorization.openshift.io/system-only': 'true'}}},
```
Comment (jwforres) [inline test/spec/services/membership/membershipSpec.js]: Can you add one with the annotation set to false

Comment (benjaminapetersen) [inline test/spec/services/membership/membershipSpec.js]: Yeah ill add a few more examples of things that wont actually pass the filter, just to be clear.

**Comment by jwforres** [comment]
[merge]

**Pull request merged**